packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -604,21 +611,57 @@ class _AppRunLogger extends Logger {
 
   @override
   void printStatus(String message, { bool emphasis: false }) {
-    domain?._sendAppEvent(app, 'log', <String, dynamic>{ 'log': message });
+    _sendLogEvent(<String, dynamic>{ 'log': message });
   }
 
   @override
   void printTrace(String message) { }
 
   @override
   Status startProgress(String message) {
-    printStatus(message);
-    return new Status();
+    int id = _nextProgressId++;
+
+    _sendLogEvent(<String, dynamic>{
+      'log': message,
+      'progress': true,
+      'id': id.toString()
+    });
+
+    return new _AppLoggerStatus(this, id);
   }
 
   void close() {
     domain = null;
   }
+
+  void _sendLogEvent(Map<String, dynamic> event) {
+    domain?._sendAppEvent(app, 'log', event);
```

Comment:
I'll log events to the flutter_tools log if they happen after the close() is called. This would be an unexpected condition however - the logs would be happening after the app has completed.